src/libcrun/cgroup-systemd.c

```diff
@@ -970,7 +999,9 @@ libcrun_destroy_cgroup_systemd (struct libcrun_cgroup_status *cgroup_status,
   if (UNLIKELY (ret < 0))
     crun_error_release (err);
 
-  return destroy_cgroup_path (cgroup_status->path, mode, err);
+  path_to_scope = get_cgroup_scope_path (cgroup_status->path, cgroup_status->scope);
+
+  return destroy_cgroup_path (path_to_scope, mode, err);
```

Comment:
@giuseppe This change breaks `crun restore` with cgroups v1:
```
$ podman pull ubi8
$ podman run -d ubi8 sleep 300
$ podman container checkpoint -l
$ podman container restore -l
Error: OCI runtime error: crun: CRIU restoring failed -52.  Please check CRIU logfile `/var/lib/containers/storage/overlay-containers/b10e1136e6cefcf97759deef7377036b6cefc0eacf9591976671712f4e3c4ac9/userdata/restore.log`
```

```
$ tail /var/lib/containers/storage/overlay-containers/b10e1136e6cefcf97759deef7377036b6cefc0eacf9591976671712f4e3c4ac9/userdata/restore.log
(00.000882) cg: 	Making controller dir .criu.cgyard.8cbQy1/freezer (freezer), type cgroup
(00.000900) cg: Created cgroup dir freezer/machine.slice/libpod-b10e1136e6cefcf97759deef7377036b6cefc0eacf9591976671712f4e3c4ac9.scope/container
(00.000904) cg: Restore special props
(00.000905) cg: 	Making controller dir .criu.cgyard.8cbQy1/cpuset (cpuset), type cgroup
(00.000929) cg: Created cgroup dir cpuset/machine.slice/libpod-b10e1136e6cefcf97759deef7377036b6cefc0eacf9591976671712f4e3c4ac9.scope/container
(00.000932) cg: Restore special props
(00.000933) cg: Restoring cgroup property value [0-15] to [cpuset/machine.slice/libpod-b10e1136e6cefcf97759deef7377036b6cefc0eacf9591976671712f4e3c4ac9.scope/container/cpuset.cpus]
(00.000939) Error (criu/cgroup.c:1490): cg: Failed writing 0-15 to cpuset/machine.slice/libpod-b10e1136e6cefcf97759deef7377036b6cefc0eacf9591976671712f4e3c4ac9.scope/container/cpuset.cpus: Permission denied
(00.000944) Error (criu/cgroup.c:1756): cg: Restoring cpuset.cpus special property failed
(00.000944) Error (criu/cgroup.c:1815): cg: Restoring special cpuset props failed!
```

The `Permission denied` error appears when writing to `cpuset/machine.slice/libpod-b10e1136e6cefcf97759deef7377036b6cefc0eacf9591976671712f4e3c4ac9.scope/container/cpuset.cpus` because `cpuset.cpus` in parent is empty.

Reply:
this change was made 2 years ago. How did it work until now?  Was it always broken?

Reply:
> How did it work until now? Was it always broken?

`crun restore` with *cgroups v1* has been broken since version 1.8.1, but it works with *cgroups v2*. We have seen this error reported in https://github.com/checkpoint-restore/criu/issues/2000 and https://issues.redhat.com/browse/RHEL-32180

Reply:
we don't really care about cgroup v1: https://github.com/containers/crun/issues/1149

support for it is going away.

Given cgroup v1 is also giving problems with https://github.com/containers/crun/issues/1589

I think I'll just open a PR to add a warning when cgroup v1 is used